library/core/src/ptr/mut_ptr.rs

```diff
@@ -404,37 +404,25 @@ impl<T: ?Sized> *mut T {
         if self.is_null() { None } else { Some(unsafe { &*(self as *const MaybeUninit<T>) }) }
     }
 
-    /// Calculates the offset from a pointer.
+    /// Adds an offset to a pointer.
     ///
     /// `count` is in units of T; e.g., a `count` of 3 represents a pointer
     /// offset of `3 * size_of::<T>()` bytes.
     ///
     /// # Safety
     ///
-    /// If any of the following conditions are violated, the result is Undefined
-    /// Behavior:
+    /// The resulting pointer's address is computed as the address of the original pointer plus the
+    /// computed offset, `count * size_of::<T>()`, with infinite precision.
```

Comment:
```suggestion
    /// computed offset, `count * size_of::<T>()` bytes.
```
I think talking about infinite precision here is more confusing than it helps. It's clearly to just say the calculation is not allowed to overflow.

Reply:
Then we'd have to spell out how exactly it is not allowed to overflow -- i.e., what the bounds are.  The nice thing about doing the arithmetic in infinite precision is that we can avoid that, since it falls out of the in-bounds requirement. But this spec is too clever for its own good...